### PR TITLE
Implement validator duties

### PR DESCRIPTION
**Pull request by ajsutton**
## PR Description
Adds the ability to produce blocks and attestations to the validator client service.

### CODE REVIEW

File: validator/client/src/main/java/tech/pegasys/artemis/validator/client/duties/AttestationProductionDuty.java [hunk 1]
```diff
@@ -13,29 +13,136 @@
...
+import tech.pegasys.artemis.util.async.SafeFuture;
+import tech.pegasys.artemis.util.bls.BLSSignature;
 import tech.pegasys.artemis.validator.api.ValidatorApiChannel;
 import tech.pegasys.artemis.validator.client.ForkProvider;
 import tech.pegasys.artemis.validator.client.Validator;
+import tech.pegasys.artemis.validator.client.signer.Signer;
 
 public class AttestationProductionDuty implements Duty {
   private static final Logger LOG = LogManager.getLogger();
+  private final Map<Integer, List<ValidatorWithCommitteePosition>> validators = new HashMap<>();
```
Comment (mbaxter) [inline validator/client/src/main/java/tech/pegasys/artemis/validator/client/duties/AttestationProductionDuty.java]:
```suggestion
  private final Map<Integer, List<ValidatorWithCommitteePosition>> validatorsByCommitteeIndex = new HashMap<>();
```

File: validator/client/src/main/java/tech/pegasys/artemis/validator/client/duties/BlockProductionDuty.java [hunk 2]
```diff
@@ -24,18 +32,52 @@
...
   private final UnsignedLong slot;
+  private final ForkProvider forkProvider;
+  private final ValidatorApiChannel validatorApiChannel;
 
   public BlockProductionDuty(
       final Validator validator,
       final UnsignedLong slot,
       final ForkProvider forkProvider,
       final ValidatorApiChannel validatorApiChannel) {
+    LOG.trace("Created");
```
Comment (mbaxter) [inline validator/client/src/main/java/tech/pegasys/artemis/validator/client/duties/BlockProductionDuty.java]: Did you mean to keep this log?

**Review by mbaxter — APPROVED** [review]
LGTM

File: validator/client/src/main/java/tech/pegasys/artemis/validator/client/duties/BlockProductionDuty.java (hunk 2 above)
Comment (ajsutton) [reply validator/client/src/main/java/tech/pegasys/artemis/validator/client/duties/BlockProductionDuty.java]: Nope, removed.

File: validator/client/src/main/java/tech/pegasys/artemis/validator/client/duties/AttestationProductionDuty.java (hunk 1 above)
Comment (ajsutton) [reply validator/client/src/main/java/tech/pegasys/artemis/validator/client/duties/AttestationProductionDuty.java]: Good idea.

**Pull request merged**